cernopendata/cold_storage/api.py

```diff
@@ -0,0 +1,206 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Open Data Portal; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Cold Storage API."""
+
+import importlib
+import logging
+from datetime import datetime
+
+from flask import current_app
+from flask_mail import Message
+from invenio_db import db
+from invenio_files_rest.models import (
+    Bucket,
+    BucketTag,
+    FileInstance,
+    ObjectVersion,
+    ObjectVersionTag,
+)
+from invenio_records_files.api import FileObject, FilesIterator, Record
+
+from .models import RequestMetadata, TransferMetadata
+
+logger = logging.getLogger(__name__)
+
+
+class FileObjectCold(FileObject):
+    """Overwrite the fileobject to get multiple URI."""
+
+    @classmethod
+    def create(self, bucket, filename, file_id):
+        """Create a cold file object."""
+        return ObjectVersion.create(bucket, filename, file_id)
+
+    def dumps(self):
+        """This one has the information about the cold URI stored in a ObjectVersionTag."""
+        info = super(FileObjectCold, self).dumps()
+        info["tags"] = {}
+        for tagName in ("uri_cold", "hot_deleted"):
+            tag = ObjectVersionTag.get(str(self.obj.version_id), tagName)
+            if tag:
+                info["tags"][tagName] = tag.value
+        if "availability" in self.data:
+            del self.data["availability"]
+        info["availability"] = self.availability
+        if "uri" not in info:
+            file = FileInstance.get(str(self.obj.file_id))
+            info["uri"] = file.uri
+        return info
+
+    @property
+    def availability(self):
+        """Describe the QoS of the file."""
+        if "availability" not in self.data:
+            avl = "ready"
+            for t in self.obj.tags:
+                if t.key == "hot_deleted":
+                    avl = "needs request"
+            self.data["availability"] = avl
+        return self.data["availability"]
+
+
+class Transfer:
+    """API for managing cold storage transfers."""
+
+    @staticmethod
+    def create(entry):
+        """Create a new transfer entry."""
+        transfer = TransferMetadata(
+            action=entry["action"],
+            new_filename=entry["new_filename"],
+            record_uuid=entry["record_uuid"],
+            file_id=entry["file_id"],
+            method=entry["method"],
+            method_id=entry.get("method_id", ""),
+            submitted=datetime.utcnow(),
+            last_check=datetime.utcnow(),
+            size=entry.get("size", None),
+        )
+        db.session.add(transfer)
+        db.session.commit()
+        return transfer
+
+    @staticmethod
+    def _get_ongoing_transfers(last_check):
```

Comment:
It seems contradictory to have a staticmethod that is private....
```suggestion
    def get_ongoing_transfers(last_check):
```